packets.go

```diff
@@ -761,40 +761,37 @@ func (rows *textRows) readRow(dest []driver.Value) error {
 	}
 
 	// RowSet Packet
-	var n int
-	var isNull bool
 	pos := 0
 
 	for i := range dest {
-		// Read bytes and convert to string
-		dest[i], isNull, n, err = readLengthEncodedString(data[pos:])
+		b, isNull, n, err := readLengthEncodedString(data[pos:])
+		if err != nil {
+			return err
+		}
 		pos += n
-		if err == nil {
-			if !isNull {
-				if !mc.parseTime {
-					continue
-				} else {
-					switch rows.rs.columns[i].fieldType {
-					case fieldTypeTimestamp, fieldTypeDateTime,
-						fieldTypeDate, fieldTypeNewDate:
-						dest[i], err = parseDateTime(
-							dest[i].([]byte),
-							mc.cfg.Loc,
-						)
-						if err == nil {
-							continue
-						}
-					default:
-						continue
-					}
-				}
 
-			} else {
-				dest[i] = nil
+		if isNull {
+			dest[i] = nil
+			continue
+		}
+		if !mc.parseTime {
+			dest[i] = b // type: []byte
+			continue
+		}
+		switch rows.rs.columns[i].fieldType {
+
+		case fieldTypeTimestamp, fieldTypeDateTime,
+			fieldTypeDate, fieldTypeNewDate:
+
+			t, err := parseDateTime(b, mc.cfg.Loc)
+			if err == nil {
+				dest[i] = t // type: time.Time
 				continue
 			}
+			// If parseDateTime failed, leave as []byte
 		}
-		return err // err != nil
```

Comment:
isn't err ignored now?
Seems like this should also have a test case.

Reply:
Okay, I just saw the comment above.
This is a semantic change, which I disagree with. We should not introduce any silent fallback behavior. The user requested a parsed time and should receive an error, not something unexpected.

Reply:
Well, it looks like I misunderstood the original code. And this shows that it must be refactored to make it easier to follow 😉 .

I will fix my code once the test suite is extended to cover that case.

Reply:
This still must be changed before I can approve this change.